src/generate.dart

```diff
@@ -166,118 +220,98 @@ class DiagramGenerator {
   }
 
   Future<Null> _optimizeImages() async {
-    Directory destDir = new Directory(path.joinAll(path.split(projectDir)
+    final Directory destDir = new Directory(path.joinAll(path.split(projectDir)
       ..removeLast()
       ..add(_diagramType)));
-    List<String> images = (await tmpDir.list().toList()).map((FileSystemEntity e) => e.path).toList();
+    final List<String> images = (await temporaryDirectory.list().toList()).map((FileSystemEntity e) => e.path).toList();
     for (String imagePath in images) {
-      FileSystemEntityType type = await FileSystemEntity.type(imagePath);
+      final FileSystemEntityType type = FileSystemEntity.typeSync(imagePath);
```

Comment:
Since this function has to be async anyhow for the directory listing, what is the advantage of using typeSync (and the other *Sync functions below)?

Reply:
See https://github.com/flutter/flutter/issues/7166 and https://github.com/dart-lang/linter/commit/42e227ad7268a569418349bc4cbbd04dcec220 -- the linter complains about these specific methods.

Reply:
OK